### PR TITLE
Migrated FilterMenu to es6 module

### PR DESCRIPTION
**Pull request by mattstrayer**
### FilterMenu ES6 Refactor

**Changes**

- Refactored for ES6
- Removed unused imports (`require`, `apphost`, `loading`, `appRouter`)

**Issues**
part of #868

### CODE REVIEW

File: src/components/filtermenu/filtermenu.js [hunk 1]
```diff
@@ -1,291 +1,302 @@
...
+}
+
+function setBasicFilter(context, key, elem) {
+    var value = elem.checked;
+    value = value ? value : null;
+    userSettings.setFilter(key, value);
+}
+
+function centerFocus(elem, horiz, on) {
+    require(['scrollHelper'], function (scrollHelper) {
```
Comment (Camc314) [inline src/components/filtermenu/filtermenu.js:166]:
```suggestion
    import('scrollHelper').then(({default: scrollHelper}) => {
```

File: src/components/filtermenu/filtermenu.js [hunk 2]
```diff
@@ -1,291 +1,302 @@
...
+            inputManager.on(elems[i], onInputCommand);
+        } else {
+            inputManager.off(elems[i], onInputCommand);
         }
     }
+}
+
+FilterMenu.prototype.show = function (options) {
+    return new Promise(function (resolve, reject) {
+        require(['text!./filtermenu.template.html'], function (template) {
```
Comment (Camc314) [inline src/components/filtermenu/filtermenu.js:227]: as above

File: src/components/filtermenu/filtermenu.js [hunk 3]
```diff
@@ -1,291 +1,302 @@
...
+            moveCheckboxFocus(e.target, 1);
+            e.preventDefault();
+            break;
+        default:
+            break;
     }
+}
 
-    function FilterMenu() {
+export function FilterMenu() {
```
Comment (Camc314) [inline src/components/filtermenu/filtermenu.js:210]: This could be converted to a class.

Comment (Camc314) [reply src/components/filtermenu/filtermenu.js:210]: And thanks for contributing to the JF Project!

Comment (mattstrayer) [reply src/components/filtermenu/filtermenu.js:210]: Happy to convert, is `components/appFooter.js` a good template to follow for class conversion @Camc314 ?

Comment (Camc314) [reply src/components/filtermenu/filtermenu.js:210]: Yeah it should be. Any question, just ask 😀.

**Comment by mattstrayer** [comment]
Looks like this is an unused component. Waiting on confirmation.

**Comment by Camc314** [comment]
> Looks like this is an unused component. Waiting on confirmation.

@dkanda @MrTimscampi can you confirm please?

**Comment by dkanada** [comment]
If you can't find any references, feel free to delete it from the codebase.

**Comment by mattstrayer** [comment]
@dkanada, @Camc314 figured out it is used in the photo list view.

File: src/components/filtermenu/filtermenu.js [hunk 4]
```diff
@@ -51,12 +65,12 @@ define(['require', 'dom', 'focusManager', 'dialogHelper', 'loading', 'apphost',
         });
 
         apiClient.getFilters(filterMenuOptions).then(function (result) {
```
Comment (MrTimscampi) [inline src/components/filtermenu/filtermenu.js:67]: You can avoid the `const self = this;` by using an arrow function, here.
```suggestion
        apiClient.getFilters(filterMenuOptions).then((result) => {
```

File: src/components/filtermenu/filtermenu.js [hunk 5]
```diff
@@ -177,40 +182,40 @@ define(['require', 'dom', 'focusManager', 'dialogHelper', 'loading', 'apphost',
...
             } else {
-                inputManager.off(elems[i], onInputCommand);
+                inputManager.off(elems[i], this.onInputCommand);
             }
         }
     }
-
-    FilterMenu.prototype.show = function (options) {
+    static show(options) {
+        const self = new FilterMenu();
```
Comment (MrTimscampi) [inline src/components/filtermenu/filtermenu.js:216]: `list.js` already instantiates FilterMenu, it's nicer to do it that way than to make this static and alias and instance, imo.

File: src/controllers/list.js [hunk 6]
```diff
@@ -407,12 +417,13 @@ define(['globalize', 'listView', 'layoutManager', 'userSettings', 'focusManager'
...
     function bindAll(elems, eventName, fn) {
-        for (var i = 0, length = elems.length; i < length; i++) {
+        for (let i = 0, length = elems.length; i < length; i++) {
             elems[i].addEventListener(eventName, fn);
         }
     }
 
-    function ItemsView(view, params) {
+class ItemsView {
+    constructor(view, params) {
```
Comment (MrTimscampi) [inline src/controllers/list.js:426]: Methods should be defined in the class and not the constructor.

File: src/controllers/list.js [hunk 7]
```diff
@@ -711,20 +722,20 @@ define(['globalize', 'listView', 'layoutManager', 'userSettings', 'focusManager'
...
         }
 
-        var i;
-        var length;
-        var btnViewSettings = view.querySelectorAll('.btnViewSettings');
+        let i;
+        let length;
+        const btnViewSettings = view.querySelectorAll('.btnViewSettings');
 
         for (i = 0, length = btnViewSettings.length; i < length; i++) {
```
Comment (MrTimscampi) [inline src/controllers/list.js:729]:
```suggestion
        for (let i = 0, length = btnViewSettings.length; i < length; i++) {
```

File: src/controllers/list.js [hunk 8]
```diff
@@ -711,20 +722,20 @@ define(['globalize', 'listView', 'layoutManager', 'userSettings', 'focusManager'
...
             btnViewSettings[i].addEventListener('click', showViewSettingsMenu.bind(this));
         }
 
-        var filterButtons = view.querySelectorAll('.btnFilter');
+        const filterButtons = view.querySelectorAll('.btnFilter');
         this.filterButtons = filterButtons;
-        var hasVisibleFilters = this.getVisibleFilters().length;
+        const hasVisibleFilters = this.getVisibleFilters().length;
 
         for (i = 0, length = filterButtons.length; i < length; i++) {
```
Comment (MrTimscampi) [inline src/controllers/list.js:737]:
```suggestion
        for (let i = 0, length = filterButtons.length; i < length; i++) {
```

**Comment by mattstrayer** [comment]
thanks, @MrTimscampi ! I'll push the changes to the `filtermenu`. I copied over @Camc314's work from the `list.js` so that I could modify for the static function change. I'll defer to that branch to make any of the suggested `list` changes.

File: src/components/filtermenu/filtermenu.js [hunk 9]
```diff
@@ -35,15 +46,14 @@ define(['require', 'dom', 'focusManager', 'dialogHelper', 'loading', 'apphost',
         elem.querySelector('.filterOptions').innerHTML = html;
     }
 
-    function renderDynamicFilters(context, result, options) {
-        renderOptions(context, '.genreFilters', 'chkGenreFilter', result.Genres, function (i) {
+    renderDynamicFilters(context, result, options) {
+        this.renderOptions(context, '.genreFilters', 'chkGenreFilter', result.Genres, function (i) {
```
Comment (Camc314) [inline src/components/filtermenu/filtermenu.js:50]: If you leave the functions that wern't previously part of ` FilterMenu.prototype` you won't need to add `this.` when the function is called

Comment (mattstrayer) [reply src/components/filtermenu/filtermenu.js]: Got it. Just pushed those changes.

File: src/components/filtermenu/filtermenu.js [hunk 10]
```diff
@@ -1,225 +1,222 @@
...
-
-        apiClient.getFilters(filterMenuOptions).then(function (result) {
-            renderDynamicFilters(context, result, options);
-        });
+}
+function saveValues(context, settings, settingsKey) {
+    var elems = context.querySelectorAll('.simpleFilter');
+    var i;
+    var length;
+    for (i = 0, length = elems.length; i < length; i++) {
```
Comment (Camc314) [inline src/components/filtermenu/filtermenu.js:107]:
```suggestion
    for (let i = 0, length = elems.length; i < length; i++) {
```

File: src/components/filtermenu/filtermenu.js (hunk 9 above)
Comment (Camc314) [reply src/components/filtermenu/filtermenu.js]: Cool thanks

File: src/components/filtermenu/filtermenu.js (hunk 10 above)
Comment (mattstrayer) [reply src/components/filtermenu/filtermenu.js:107]: great catch, I'll get that fixed in a minute

Comment (Camc314) [reply src/components/filtermenu/filtermenu.js]: Welp this is my mistake, `let` needs to be added to all the other for loops

**Pull request merged**